src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+    private final Random random = new Random();
+    private final ColorSupplier supplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(NUMBER_OF_FIGURES);
+
+        switch (figureType) {
+            case 0:
+                String randomColorForSquare = supplier.getRandomColor();
```

Comment:
move generating random color outside from switch clauses, it's unjustified code duplication. Each time we enter the method we will have a new random color nevertheless.

Reply:
i do this, but now i get the same color for each figure